pkgs/stdenv/generic/setup.sh

```diff
@@ -62,7 +63,8 @@ _callImplicitHook() {
     local hookName="$2"
     if declare -F "$hookName" > /dev/null; then
         "$hookName"
-    elif type -p "$hookName" > /dev/null; then
+    elif test -e "$hookName" ; then
```

Comment:
this is not the same thing:
```
$ type -p dirname && echo OK || echo FAIL
/run/current-system/sw/bin/dirname
OK
$ test -e dirname && echo OK || echo FAIL
FAIL
```
I'm only looking at the diff so maybe there is an explanation in the git history.

Reply:
no you're right. I only thought about a file case. 
My problem was that type -p return a non-zero exit code on failure.
The reason I thought about a file only is that inside the conditional is a source. You can't source a dir right?